crates/ruff_linter/src/rules/flake8_markupsafe/rules/unsafe_markup_use.rs

```diff
@@ -0,0 +1,100 @@
+use ruff_python_ast::ExprCall;
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::name::QualifiedName;
+use ruff_text_size::Ranged;
+
+use crate::{checkers::ast::Checker, settings::LinterSettings};
+
+/// ## What it does
+/// Checks for non-literal strings being passed to [`markupsafe.Markup`].
+///
+/// ## Why is this bad?
+/// `markupsafe.Markup` does not perform any escaping, so passing dynamic
+/// content, like f-strings, variables or interpolated strings will potentially
+/// lead to XSS vulnerabilities.
+///
+/// Instead you should interpolate the [`markupsafe.Markup`] object.
+///
+/// Using [`lint.flake8-markupsafe.extend-markup-names`] additional objects
+/// can be treated like [`markupsafe.Markup`].
+///
+/// ## Example
+/// ```python
+/// from markupsafe import Markup
+///
+/// content = "<script>alert('Hello, world!')</script>"
+/// html = Markup(f"<b>{content}</b>")  # XSS
+/// ```
+///
+/// Use instead:
+/// ```python
+/// from markupsafe import Markup
+///
+/// content = "<script>alert('Hello, world!')</script>"
+/// html = Markup("<b>{}</b>").format(content)  # Safe
+/// ```
+///
+/// ## References
+/// - [MarkupSafe](https://pypi.org/project/MarkupSafe/)
+/// - [`markupsafe.Markup`](https://markupsafe.palletsprojects.com/en/stable/escaping/#markupsafe.Markup)
+///
+/// [markupsafe.Markup]: https://markupsafe.palletsprojects.com/en/stable/escaping/#markupsafe.Markup
+#[violation]
+pub struct UnsafeMarkupUse {
+    name: String,
+}
+
+impl Violation for UnsafeMarkupUse {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let UnsafeMarkupUse { name } = self;
+        format!("Unsafe use of `{name}` detected")
+    }
+}
+
+/// Checks for unsafe calls to `[markupsafe.Markup]`.
+///
+/// [markupsafe.Markup]: https://markupsafe.palletsprojects.com/en/stable/escaping/#markupsafe.Markup
+pub(crate) fn unsafe_markup_call(checker: &mut Checker, call: &ExprCall) {
+    if let Some(name) = checker
```

Comment:
To only consider this rule if `markupsafe` or `flask` is imported:

```suggestion
     if !(checker.semantic().seen_module(Modules::MARKUPSAFE)
        || checker.semantic().seen_module(Modules::FLASK))
    {
        return;
    }
    if let Some(name) = checker
```

(And add `MARKUPSAFE` and `FLASK` to `Modules`)

Reply:
This fast-path only works if no additional names have been specified in the options, so at the very least I would have to first check `checker.settings.flake8_markupsafe.extend_markup_names.is_empty()`.

Reply:
That's true. Such a guard would be helpful for users that do not work with `markupsafe` or `flask` but have this rule, but have it enabled.